File: 900/1216B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int n;
    cin>>n;

    vector<int> a(n);
    for (int i=0; i<n; i++)
        cin>>a[i];

    // Store indices to keep track of original positions (1-based later)
    vector<int> idx(n);
    iota(idx.begin(), idx.end(), 0);

    // Sort indices descending based on the values in 'a'
    sort(idx.begin(), idx.end(), [&](int i, int j) {
        return a[i] > a[j];
    });

    long long shots = 0;
    for (int k=0; k<n; k++)
        shots += (1LL * a[idx[k]] * k) + 1;

    cout<<shots<<"\n";

    for(int i=0; i<n; i++)
        cout<<(idx[i] + 1) << (i == n - 1 ? "" : " ");

    cout << "\n";

    return 0;
}

File: 900/1216B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
 
        int n = sc.nextInt();
        int a[] = new int[n];
 
        for(int i=0;i<n;i++)
            a[i] = sc.nextInt();
 
        Integer idx[] = new Integer[n];
        for(int i=0;i<n;i++)
            idx[i] = i;
 
        Arrays.sort(idx,(i,j) -> a[j]-a[i]);
        long shots = 0;
 
        for(int k=0;k<n;k++)
            shots += (long)a[idx[k]]*k+1;
 
        System.out.println(shots);
        for(int i=0;i<n;i++)
            System.out.print((idx[i]+1)+" ");
    }
}